src/network/networkd-link.c

```diff
@@ -1692,12 +1692,17 @@ static int link_configure_addrgen_mode(Link *link) {
 
         if (!link_ipv6ll_enabled(link))
                 ipv6ll_mode = IN6_ADDR_GEN_MODE_NONE;
-        else if (sysctl_read_ip_property(AF_INET6, link->ifname, "stable_secret", NULL) < 0)
-                /* The file may not exist. And even if it exists, when stable_secret is unset,
-                 * reading the file fails with EIO. */
-                ipv6ll_mode = IN6_ADDR_GEN_MODE_EUI64;
-        else
-                ipv6ll_mode = IN6_ADDR_GEN_MODE_STABLE_PRIVACY;
+        else if (link->network->ipv6_address_gen_mode == _LINK_IPV6_ADDRESS_GEN_MODE_INVALID) {
```

Comment:
doesn't matter at all, but instead of an explicit == _LINK_IPV6_ADDRESS_GEN_MOVE_INVALID compariso, I'd probably have checked < 0 here, and consider any negative value as invalid